docs/guide/connecting.asciidoc

```diff
@@ -3,58 +3,155 @@
-])
-----------------------------
+NOTE: If you don't specify `ca_certs` or `ssl_assert_fingerprint` then the https://certifiio.readthedocs.io[certifi package] will be used for `ca_certs` by default if available.
 
 [discrete]
-[[connect-ec]]
-==== Connecting to Elastic Cloud
+==== Verifying HTTPS with certificate fingerprints
 
-Cloud ID is an easy way to configure your client to work with your Elastic Cloud 
-deployment. Combine the `cloud_id` with either `basic_auth` or `api_key` to 
-authenticate with your Elastic Cloud deployment.
+NOTE: Using this method **requires using Python 3.10 or later** and isn't available when using the `aiohttp` HTTP client library so can't be used with `AsyncElasticsearch`.
 
-Using `cloud_id` enables TLS verification and HTTP compression by default and 
-sets the port to 443 unless otherwise overwritten via the port parameter or the 
-port value encoded within `cloud_id`. Using Cloud ID also disables sniffing as
-a proxy is in use.
+This method of verifying the HTTPS connection takes advantage of the certificate fingerprint value noted down earlier.
 
-[source,python]
-----------------------------
+If you don't have access to the CA fingerprint value from the Elasticsearch startup phase you can use the following script to output the root CA fingerprint of the Elasticsearch instance running on `localhost` at port `9200`:
+
+```bash
+# Replace the values of 'localhost' and '9200' to the
+# corresponding host and port values for the cluster.
+openssl s_client -connect localhost:9200 -servername localhost -showcerts </dev/null 2>/dev/null \
+  | openssl x509 -in /dev/stdin -sha256 -noout -fingerprint
+```
+
+The output of `openssl x509` will look something like this:
+
+```
+SHA256 Fingerprint=A5:2D:D9:35:11:E8:C6:04:5E:21:F1:66:54:B7:7C:9E:E0:F3:4A:EA:26:D9:F4:03:20:B5:31:C4:74:67:62:28
+```
+
+Take this SHA256 fingerprint value and pass it to the Python Elasticsearch client via `ssl_assert_fingerprint`:
+
+```python
 from elasticsearch import Elasticsearch
 
-es = Elasticsearch(
-    cloud_id="cluster-1:dXMa5Fx..."
+# Fingerprint either from Elasticsearch startup or above script
+CERT_FINGERPRINT = "A5:2D:D9:35:11:E8:C6:04:5E:21:F1:66:54:B7:7C:9E:E0:F3:4A:EA:26:D9:F4:03:20:B5:31:C4:74:67:62:28"
+
+# Generated by Elasticsearch
+ELASTIC_PASSWORD = "<password>"
+
+client = Elasticsearch(
+  "https://localhost:9200",
+  # Colons and uppercase/lowercase don't matter
+  # for 'ssl_assert_fingerprint'.
+  ssl_assert_fingerprint=CERT_FINGERPRINT,
+  basic_auth=("elastic", ELASTIC_PASSWORD)
 )
-----------------------------
+
+# Successful response!
+client.info()
+# {'name': 'instance-0000000000', 'cluster_name': ...}
+```
+
+[discrete]
+[[connect-no-security]]
+=== Connecting without security enabled
+
+WARNING: Running Elasticsearch without security enabled is strongly recommended against as it leaves your cluster open to data breaches, malware, and ransomware attacks.
```

Comment:
Is there an Elasticsearch docs page explaining how to disable security that we could link to?

Reply:
I was thinking something similar: that we should link to a page on how to do this, and what the consequences of doing so are, so that users are free to make an informed choice. Probably this is better to discuss in a centralised document, rather than a client manual.